compute/standard-deviation/docs/modules/ROOT/pages/compute-standard-deviation.adoc

```diff
@@ -0,0 +1,332 @@
+
+Now imagine the you have 1,000,000 `Customer` objects stored on two Hazelcast members. One member could handle this volume but wouldn't be resilient to failure. So the junior developer approach now requests 1,000,000 integers from the cluster across the network and the senior developer approach retrieves two doubles, one from each member.
+
+It's clear that if data volumes increase, the cluster-side computation copes better. The same amount of data still has to be examined, but less data moves.
+
+.Client `JuniorDeveloper.java`
+[%collapsible]
+====
+In this class, the average is calculated like this:
+
+[source,java,indent=0]
+----
+include::standard-deviation:example$client/src/main/java/your/company/name/JuniorDeveloper.java[tag=average]
+----
+
+This code is easy to understand, which is good for maintenance. However, it has three flaws:
+
+- This code requires every customer record to be moved from where the data is stored (in this case the Hazelcast cluster) to where the calculation is run. Even if a projection were added, this is very heavy on the network.
+
+- The `keySet()` operation produces a collection that contains all the keys to iterate across. If this collection is large, it could result in memory overflow.
+
+- If the map is empty, the code results in a division by zero. A good unit test would find this.
+
+The sum of the square of the differences is calculated on the client-side like this:
+
+[source,java,indent=0]
+----
+include::standard-deviation:example$client/src/main/java/your/company/name/JuniorDeveloper.java[tag=stdev]
+----
+
+link:{github-directory}/client/src/main/java/your/company/name/JuniorDeveloper.java[View source]
+====
+
+.Client `SeniorDeveloper.java`
+[%collapsible]
+====
+In this class, the average is calculated in one line:
+
+[source,java,indent=0]
+----
+include::standard-deviation:example$client/src/main/java/your/company/name/SeniorDeveloper.java[tag=average]
+----
+
+Hazelcast provides built-in functions for this type of calculation.
+
+The sum of the square of the differences is calculated on the cluster-side like this:
+
+[source,java,indent=0]
+----
+include::standard-deviation:example$client/src/main/java/your/company/name/SeniorDeveloper.java[tag=stdev]
+----
+
+The client uses the ExecutorService API to run a `Callable` task on each member in the cluster. The `Callable` class is named `TotalDifferenceSquaredCallable.java` and it runs the calculation on the Hazelcast member where it is invoked.
+
+Each member calculates a sub-total only for the data that member owns. If you have two members, the calculation takes half the time. If you have ten members, the calculation takes a tenth of the time. Each member runs it's calculation independently, the runtime is dependent on how much data each member hosts rather than how much data as a whole exists.
```

Comment:
```suggestion
Each member calculates a subtotal only for the data that member owns. If you have two members, the calculation takes half the time. If you have ten members, the calculation takes a tenth of the time. Each member runs its calculation independently, the runtime is dependent on how much data each member hosts rather than how much data exists as a whole.
```